client/src/components/Applet/Applet.js

```diff
@@ -91,11 +277,37 @@ class Applet extends Component {
     this.setState({ isCalendarDisplayed: !currentState });
   }
 
+  handleExistingBookings() {
+    let existingBookings = this.state.roomData.bookings;
+    let bookedDates = [];
+
+    for (let i = 0; i < existingBookings.length; i++) {
+      let duration = existingBookings[i].duration;
+      bookedDates.push(new Date(existingBookings[i].checkIn));
+      for (let j = 1; j < duration + 0; j++) {
```

Comment:
You may want to double check the for loop and adding hard coded 0 to the duration.